### PR TITLE
Fix Linux unit tests

### LINKED ISSUE #86: Fix and enable tests in CI for Linux
While working on #85, I've ran the tests on Linux to validate my changes work on both Windows and Linux, however on Linux the tests are failing:

```
Failed!  - Failed:    13, Passed:   415, Skipped:     0, Total:   428, Duration: 2 s - Zio.Tests.dll (net8.0)
```

The following tests are failing:

```
[xUnit.net 00:00:00.44]     Zio.Tests.FileSystems.TestPhysicalFileSystemCompat.TestDirectoryDeleteAndOpenFile [FAIL]
[xUnit.net 00:00:00.57]     Zio.Tests.FileSystems.TestPhysicalFileSystem.TestFile [FAIL]
[xUnit.net 00:00:00.70]     Zio.Tests.FileSystems.TestZipArchiveFileSystem.TestCommonRead [FAIL]
[xUnit.net 00:00:00.71]     Zio.Tests.FileSystems.TestPhysicalFileSystem.TestDirectory [FAIL]
[xUnit.net 00:00:00.83]     Zio.Tests.FileSystems.TestSubFileSystem.TestBasic [FAIL]
[xUnit.net 00:00:00.91]     Zio.Tests.FileSystems.TestPhysicalFileSystem.TestFileExceptions [FAIL]
[xUnit.net 00:00:01.10]     Zio.Tests.FileSystems.TestPhysicalFileSystem.TestCommonRead [FAIL]
[xUnit.net 00:00:01.35]     Zio.Tests.FileSystems.TestPhysicalFileSystem.TestDirectoryExceptions [FAIL]
[xUnit.net 00:00:01.37]     Zio.Tests.FileSystems.TestPhysicalFileSystemCompat.TestDirectoryExceptions [FAIL]
[xUnit.net 00:00:01.73]     Zio.Tests.FileSystems.TestPhysicalFileSystem.TestEnumerate [FAIL]
[xUnit.net 00:00:01.76]     Zio.Tests.FileSystems.TestPhysicalFileSystemCompat.TestFile [FAIL]
[xUnit.net 00:00:01.79]     Zio.Tests.FileSystems.TestPhysicalFileSystem.TestFileSystemInvalidDriveLetter [FAIL]
[xUnit.net 00:00:02.35]     Zio.Tests.FileSystems.TestPhysicalFileSystemCompat.TestFileExceptions [FAIL]
```

GitHub Actions doesn't catch this because the CI only runs on Windows:

https://github.com/xoofx/zio/blob/c23f0a0b822089f14a713482178efe47ce751151/.github/workflows/ci.yml#L14

### PR DESCRIPTION
**Pull request by GerardSmit**
Fixes #86

### CODE REVIEW

File: src/Zio/FileSystems/MemoryFileSystem.cs [hunk 1]
```diff
@@ -1744,7 +1744,8 @@ public FileNode(MemoryFileSystem fileSystem, DirectoryNode parentNode, string? n
             }
             else
             {
-                Attributes = FileAttributes.Archive;
+                // Mimic OS-specific attributes.
+                Attributes = PhysicalFileSystem.IsOnWindows ? FileAttributes.Archive : FileAttributes.Normal;
```
Comment (GerardSmit) [inline src/Zio/FileSystems/MemoryFileSystem.cs:1748]: Breaking change: Linux doesn't have the `FileAttributes.Archive` attribute, so I modified the MemoryFileSystem to mimic the OS filesystem. Is this OK?

File: src/Zio/FileSystems/PhysicalFileSystem.cs [hunk 2]
```diff
@@ -23,9 +23,9 @@ public class PhysicalFileSystem : FileSystem
     private const string DrivePrefixOnWindows = "/mnt/";
     private static readonly UPath PathDrivePrefixOnWindows = new UPath(DrivePrefixOnWindows);
 #if NETSTANDARD
-    private static readonly bool IsOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+    internal static readonly bool IsOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
 #else
-    private static readonly bool IsOnWindows = CheckIsOnWindows();
+    internal static readonly bool IsOnWindows = CheckIsOnWindows();
```
Comment (GerardSmit) [inline src/Zio/FileSystems/PhysicalFileSystem.cs:28]: Changed to internal for the OS-check in [MemoryFileSystem.cs](https://github.com/xoofx/zio/pull/87/files#diff-3100cc6fb29e710fbafa4f0cab7a2dae0998f0005393f5631ab9d820f2088b3c)

File: src/Zio/FileSystems/ReadOnlyFileSystem.cs [hunk 3]
```diff
@@ -102,7 +102,11 @@ protected override Stream OpenFileImpl(UPath path, FileMode mode, FileAccess acc
     protected override FileAttributes GetAttributesImpl(UPath path)
     {
         // All paths are readonly
-        return base.GetAttributesImpl(path) | FileAttributes.ReadOnly;
+        var attributes = base.GetAttributesImpl(path);
+
+        return attributes == FileAttributes.Normal
+            ? FileAttributes.ReadOnly
```
Comment (GerardSmit) [inline src/Zio/FileSystems/ReadOnlyFileSystem.cs:108]: Breaking change: Normal is used when there are no attributes. Combining "Normal" and "ReadOnly" isn't valid, so I changed it so it only returns ReadOnly.

File: src/Zio/FileSystems/AggregateFileSystem.cs [hunk 4]
```diff
@@ -287,7 +287,11 @@ protected override Stream OpenFileImpl(UPath path, FileMode mode, FileAccess acc
     protected override FileAttributes GetAttributesImpl(UPath path)
     {
         var entry = GetPath(path);
-        return entry.FileSystem.GetAttributes(path) | FileAttributes.ReadOnly;
+        var attributes = entry.FileSystem.GetAttributes(path);
+
+        return attributes == FileAttributes.Normal
+            ? FileAttributes.ReadOnly
```
Comment (GerardSmit) [inline src/Zio/FileSystems/AggregateFileSystem.cs:293]: Breaking change: Normal is used when there are no attributes. Combining "Normal" and "ReadOnly" isn't valid, so I changed it so it only returns ReadOnly.

File: src/Zio.Tests/FileSystems/TestPhysicalFileSystem.cs [hunk 5]
```diff
@@ -92,15 +94,20 @@ public void TestDirectory()
             // LastAccessTime
             // LastWriteTime
             // CreationTime
-            var lastWriteTime = DateTime.Now + TimeSpan.FromSeconds(10);
-            var lastAccessTime = DateTime.Now + TimeSpan.FromSeconds(11);
-            var creationTime = DateTime.Now + TimeSpan.FromSeconds(12);
+            if (IsWindows)
+            {
+                var creationTime = new DateTime(2000, 1, 1, 0, 0, 0, DateTimeKind.Local);
```
Comment (GerardSmit) [inline src/Zio.Tests/FileSystems/TestPhysicalFileSystem.cs:99]: Changed the test dates so it was easier to see which one was failing.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -26,7 +30,15 @@ jobs:
         dotnet-version: '8.0.x'
 
     - name: Build, Test, Pack, Publish
+      if: matrix.os == 'windows-latest'
+      shell: bash
+      run: |
+        dotnet tool install -g dotnet-releaser --configfile .github/workflows/nuget_org_only.config
+        dotnet-releaser run --nuget-token "${{secrets.NUGET_TOKEN}}" --github-token "${{secrets.GITHUB_TOKEN}}" src/dotnet-releaser.toml
+
+    - name: Build, Test
```
Comment (GerardSmit) [inline .github/workflows/ci.yml:39]: Not sure if you want to create a new .yml-file for Linux or use a matrix here.

File: src/Zio/FileSystems/ZipArchiveFileSystem.cs [hunk 7]
```diff
@@ -492,12 +492,15 @@ protected override FileAttributes GetAttributesImpl(UPath path)
         var attributes = entry.FullName[entry.FullName.Length - 1] == DirectorySeparator ? FileAttributes.Directory : 0;
 
 #if NETSTANDARD2_1_OR_GREATER || NET6_0_OR_GREATER
-        if (entry.ExternalAttributes == 0 && attributes == 0)
+        const FileAttributes validValues = (FileAttributes)0x7FFF /* Up to FileAttributes.Encrypted */ | FileAttributes.IntegrityStream | FileAttributes.NoScrubData;
```
Comment (GerardSmit) [inline src/Zio/FileSystems/ZipArchiveFileSystem.cs:495]: On Linux the `entry.ExternalAttributes` was `-2119958496`, without this I couldn't make out what the attributes were...

File: src/Zio.Tests/Zio.Tests.csproj [hunk 8]
```diff
@@ -9,11 +9,12 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
     <PackageReference Include="xunit" Version="2.6.3" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.5">
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.6">
```
Comment (GerardSmit) [inline src/Zio.Tests/Zio.Tests.csproj:12]: Fixes https://github.com/xunit/visualstudio.xunit/issues/396

**Review by GerardSmit — COMMENTED** [review]
Some remarks why I changed some code in `src/`

**Comment by xoofx** [comment]
Thanks!

**Pull request merged**